src/System Application/App/Format Validation/src/FormatValidation.Codeunit.al

```diff
@@ -0,0 +1,38 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Utilities;
+
+/// <summary>
+/// Codeunit that exposes format validation functionality.
+/// </summary>
+codeunit 348 "Format Validation"
+{
+    Access = Public;
+
+    /// <summary>
+    /// Validates if a phone no. contains characters.
+    /// </summary>
+    /// <param name="PhoneNo">The phone no. to validate</param>
+    procedure ValidateIfPhoneNoContainsCharacters(PhoneNo: Text[30]): Boolean
```

Comment:
@nikolakukrika, I requested an additional check on field phone no. for table contact alt. address similar to the format validation for contact phone no. I understand that the module should be extended in time, but that's out of scope. The module is created, and it up to everyone who wants to contribute to extend it. My boundaries are reached.

Reply:
@DonHendriksen, this small PR has led to rather intense discussions internally. Some love the idea of having a format validation module, others don't. Some think your PR is fine, some strongly disagree.

The biggest push back is on the fact, that your module has too narrow a scope, which might lead to us not getting the design right. It's not that all kinds of format validations must be implemented now, but they should be considered to nail down a future proof design. Sure, we could go for the exact thing you need right here, right now, but then it probably should not be a system application module, as those typically are supposed to have a broader applicability.

Let's be more concrete: In essence, your function validates that a phone no. only contains numbers. Why is that specific to a phone no? Why not make a general validation, that a string doesn't contain characters? This might come in handy other places!

But what if someone actually wants to allow to store a phone no. like 1-800-CALLMENOW, which is seen in some countries. How does one now override the behavior? There is no interface where one could change the implementation. There are no events either - which we'd also ideally would like to avoid, now that we're designing a new module. Sure, that also doesn't exist today, but does that mean a new module shouldn't support that?

I was much in favor of a format validation module, but maybe building it around a single format validation with hardly any requirements to other formats doesn't work. It also seems like we're asked to wildly over-engineer, what initially was a simple module.

I'm still between chairs, but I'm tempted to go back on the module and return to your original implementation in the BaseApp - maybe even simplifying that one even more.

Reply:
I think the worst what could happen is that this PR is cancelled.
It would be better if the critics provide help in making this PR to something really helpful.
If we need an interface for this -> then help in providing a concept for this. If I knew a good example for extensibility or an "Inteface-Handled" pattern then we could uptake this pattern here, but nothing comes to my mind.
What is a generic interface...?

To be honest I never understood why the mail validation mas done in the mail account codeunit.
https://github.com/microsoft/BCApps/blob/main/src/System%20Application/App/Email/src/Account/EmailAccount.Codeunit.al
We could move this a generic Format Validation Module.

Reply:
Hi @JesperSchulz, it is quit around here. Can we discuss what will be the final solution?

Reply:
@DonHendriksen Jesper was last two weeks on vacation and he will be one more week on vacation, that's why it's quiet. He will probably get back to you upon returning when he catches up all what was happening :)

Reply:
@JesperSchulz, I guess all of us enjoyed a holiday. Now we are all back I really like to proceed with this topic. How can we come to a final conclusion for the phone no. check?

The fact is that there are clear restrictions for Dutch phone nos., but these cannot be applied as a default check because other countries have different restrictions for phone nos. So it should be something country specific.
Or, as you considered, going back to where I started with.

Can you, or can other contributors, draw a final conclusion on this?

Thanks and regards,
Don

Reply:
@DonHendriksen, I agree. Time to reach a conclusion. Let's go with your original solution, and I will work on a format module incl. refactoring when time allows. We need to have some more design discussions before we embark on this adventure. I still think a formatting module like the one you started here would be a very valuable addition to the product, but it requires an initial investment I cannot set aside time for right now. This work won't be lost - we'll simply park it for the time being.